File: app/src/main/java/com/cielo/ordermanager/sdk/sample/PaymentActivity.java
package com.cielo.ordermanager.sdk.sample;


import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Toast;

import androidx.annotation.NonNull;

import com.cielo.ordermanager.sdk.R;
import com.cielo.ordermanager.sdk.adapter.PaymentCodeSpinnerAdapter;

import java.util.Arrays;
import java.util.List;

import cielo.orders.domain.CheckoutRequest;
import cielo.orders.domain.Order;
import cielo.orders.domain.SubAcquirer;
import cielo.sdk.order.payment.PaymentError;
import cielo.sdk.order.payment.PaymentListener;

public class PaymentActivity extends BasePaymentActivity {



    @Override
    protected void configUi() {
        super.configUi();

        productName = "Teste - Pagamentos";

        orderData.setVisibility(View.VISIBLE);
        paymentData.setVisibility(View.GONE);

        try {
            paymentCodeAdapter = new PaymentCodeSpinnerAdapter(this, R.layout.spinner_item);
            primarySpinner.setAdapter(paymentCodeAdapter);

            primarySpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {

                @Override
                public void onItemSelected(AdapterView<?> adapterView, View view, int position,
                                           long id) {
                    paymentCode = paymentCodeAdapter.getItem(position);
                }

                @Override
                public void onNothingSelected(AdapterView<?> adapter) {
                }
            });

            List<String> installmentsArray = Arrays.asList(getResources()
                    .getStringArray(R.array.installments_array));
            final ArrayAdapter<String> installmentsAdapter =
                    new ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item,
                            installmentsArray);

            installmentsSpinner.setAdapter(installmentsAdapter);
            contentSecondary.setVisibility(View.GONE);

            installmentsSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
                @Override
                public void onItemSelected(AdapterView<?> adapterView,
                                           View view, int position, long id) {
                    installments = Integer.parseInt(installmentsAdapter.getItem(position));
                }

                @Override
                public void onNothingSelected(AdapterView<?> adapter) {
                }
            });

        } catch (UnsupportedOperationException e) {
            Toast.makeText(this, "Essa funcionalidade não está disponível nessa versão da Lio.",
                    Toast.LENGTH_SHORT).show();
        }
    }

    @Override
    public void makePayment() {
        if (order != null) {
            String ec = merchantCode.getText().toString();
            String userEmail = email.getText().toString();

            CheckoutRequest.Builder requestBuilder = new CheckoutRequest.Builder()
                    .orderId(order.getId())
                    .amount(order.pendingAmount())
                    .paymentCode(paymentCode)
                    .installments(installments);

            if (cbSubAcquirer.isChecked()) {
                if (validateSubAcquirerFields()) {
                    requestBuilder.subAcquirer(new SubAcquirer(
                            softDescriptorSub.getText().toString(),
                            terminalIdSub.getText().toString(),
                            merchantCode.getText().toString(),
                            citySub.getText().toString(),
                            telephoneSub.getText().toString(),
                            stateSub.getText().toString(),
                            postalCodeSub.getText().toString(),
                            addressSub.getText().toString(),
                            idSub.getText().toString(),
                            mccSub.getText().toString(),
                            countrySub.getText().toString(),
                            informationTypeSub.getText().toString(),
                            documentSub.getText().toString(),
                            ibgeSub.getText().toString()
                    ));
                } else {
                    Toast.makeText(this, "Necessário preencher todos os campos do subadquirente.",
                            Toast.LENGTH_SHORT).show();
                    return;
                }
            }

            if (!ec.equals(""))
                requestBuilder.ec(ec);

            if (!userEmail.equals(""))
                requestBuilder.email(userEmail);

            CheckoutRequest request = requestBuilder.build();

            orderManager.checkoutOrder(request, new PaymentListener() {

                @Override
                public void onStart() {
                    Log.d(TAG, "ON START");
                }

                @Override
                public void onPayment(@NonNull Order paidOrder) {
                    Log.d(TAG, "ON PAYMENT");

                    order = paidOrder;

                    order.markAsPaid();
                    orderManager.updateOrder(order);

                    Toast.makeText(PaymentActivity.this,"Pagamento efetuado com sucesso.",
                            Toast.LENGTH_LONG).show();

                    resetState();
                }

                @Override
                public void onCancel() {
                    Log.d(TAG, "ON CANCEL");
                    resetState();
                }

                @Override
                public void onError(@NonNull PaymentError paymentError) {
                    Log.d(TAG, "ON ERROR");
                    resetState();
                }

            });
        }
    }

    private boolean validateSubAcquirerFields() {
        return !softDescriptorSub.getText().toString().isEmpty() &&
                !terminalIdSub.getText().toString().isEmpty() &&
                !merchantCode.getText().toString().isEmpty() &&
                !citySub.getText().toString().isEmpty() &&
                !telephoneSub.getText().toString().isEmpty() &&
                !stateSub.getText().toString().isEmpty() &&
                !postalCodeSub.getText().toString().isEmpty() &&
                !addressSub.getText().toString().isEmpty() &&
                !idSub.getText().toString().isEmpty() &&
                !mccSub.getText().toString().isEmpty() &&
                !countrySub.getText().toString().isEmpty() &&
                !informationTypeSub.getText().toString().isEmpty();
    }

    @Override
    protected void onDestroy() {
        orderManager.unbind();
        super.onDestroy();
    }
}

File: app/src/main/java/com/cielo/ordermanager/sdk/sample/PaymentActivity.kt
package com.cielo.ordermanager.sdk.sample

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemSelectedListener
import android.widget.ArrayAdapter
import android.widget.Toast
import cielo.orders.domain.CheckoutRequest
import cielo.orders.domain.Order
import cielo.sdk.order.payment.PaymentError
import cielo.sdk.order.payment.PaymentListener
import com.cielo.ordermanager.sdk.R
import com.cielo.ordermanager.sdk.TAG
import com.cielo.ordermanager.sdk.adapter.PaymentCodeSpinnerAdapter
import com.cielo.ordermanager.sdk.sample.PaymentActivity
import kotlinx.android.synthetic.main.activity_payment.contentSecondary
import kotlinx.android.synthetic.main.activity_payment.email
import kotlinx.android.synthetic.main.activity_payment.installmentsSpinner
import kotlinx.android.synthetic.main.activity_payment.merchantCode
import kotlinx.android.synthetic.main.activity_payment.primarySpinner
import java.util.Arrays

class PaymentActivity : BasePaymentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        configSDK(this::onServiceBound)

    }

    override fun configUi() {
        super.configUi()
        productName = "Teste - Pagamentos"
        try {
            paymentCodeAdapter = PaymentCodeSpinnerAdapter(this, R.layout.spinner_item)
            primarySpinner.adapter = paymentCodeAdapter
            primarySpinner.onItemSelectedListener = object : OnItemSelectedListener {
                override fun onItemSelected(adapterView: AdapterView<*>?, view: View, position: Int,
                                            id: Long) {
                    paymentCode = paymentCodeAdapter?.getItem(position)
                }

                override fun onNothingSelected(adapter: AdapterView<*>?) {}
            }
            val installmentsArray = Arrays.asList(*resources
                    .getStringArray(R.array.installments_array))
            val installmentsAdapter: ArrayAdapter<String?> = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item,
                    installmentsArray)
            installmentsSpinner.adapter = installmentsAdapter
            contentSecondary.visibility = View.GONE
            installmentsSpinner.onItemSelectedListener = object : OnItemSelectedListener {
                override fun onItemSelected(adapterView: AdapterView<*>?,
                                            view: View, position: Int, id: Long) {
                    installments = installmentsAdapter.getItem(position)?.toInt() ?: 0
                }

                override fun onNothingSelected(adapter: AdapterView<*>?) {}
            }
        } catch (e: UnsupportedOperationException) {
            Toast.makeText(this, "Essa funcionalidade não está disponível nessa versão da Lio.",
                    Toast.LENGTH_SHORT).show()
        }
    }

    override fun makePayment() {
        order?.let { order ->
            orderManager?.placeOrder(order)
            val ec = merchantCode.text.toString()
            val userEmail = email.text.toString()
            val requestBuilder = paymentCode?.let {
                CheckoutRequest.Builder()
                        .orderId(order.id)
                        .amount(itemValue)
                        .paymentCode(it)
                        .installments(installments)
            }
            if (ec != "") requestBuilder?.ec(ec)
            if (userEmail != "") requestBuilder?.email(userEmail)
            val request = requestBuilder?.build()
            orderManager?.checkoutOrder(request ?: return@let, object : PaymentListener {
                override fun onStart() {
                    Log.d(TAG, "ON START")
                }

                override fun onPayment(paidOrder: Order) {
                    Log.d(TAG, "ON PAYMENT")
                    this@PaymentActivity.order = paidOrder
                    order.markAsPaid()
                    orderManager?.updateOrder(order)
                    Toast.makeText(this@PaymentActivity, "Pagamento efetuado com sucesso.",
                            Toast.LENGTH_LONG).show()
                    resetState()
                }

                override fun onCancel() {
                    Log.d(TAG, "ON CANCEL")
                    Toast.makeText(this@PaymentActivity, "Pagamento cancelado.",
                            Toast.LENGTH_LONG).show()
                    resetState()
                }

                override fun onError(paymentError: PaymentError) {
                    Log.d(TAG, "ON ERROR")
                    Toast.makeText(this@PaymentActivity, "Houve um erro: ${paymentError.description}",
                            Toast.LENGTH_LONG).show()
                    resetState()
                }
            })
        }
    }

    private fun onServiceBound() {
        orderManager?.createDraftOrder("REFERENCIA DA ORDEM")
    }
}
